R/friendly-type.R

```diff
@@ -0,0 +1,186 @@
+
+# This is based on rlang:::obj_type_friendly, but adapted to cli
```

Comment:
Yes, it uses rlang, and I also want to use cli styles.

Reply:
Can you wrap around rlang's `obj_type_friendly()` function to add cli styles? I think it's important to have a single source for this function.

Reply:
I don't think it is possible to wrap it.

Reply:
Maybe we could have some tests to make sure that they behave the same? OTOH cli already behaves differently for data frames...

Reply:
yeah and that data frame behaviour has not been ported upstream :-(
This is why I am worried about having multiple sources for that function. It would be nice to find a solution.

Reply:
Once the file in rlang does not depend on rlang, let me know and then we can use it in cli.

Reply:
I was thinking that rlang function would become less and less important because most errors would use cli.

Reply:
Many errors are thrown from rlang so we need a compat file either way, and it would be good to have only one source of implementation.